audio_analysis.py

```diff
@@ -17,132 +17,75 @@ def get_wav_properties(wav_path):
-            silence_durations_short.append(k[0] - seg_limits_short[p][1])
-        word_speech_short = k[1] - k[0]
-        total_speech_short = total_speech_short + word_speech_short
-        counter = counter + 1
-    speech_ratio_short = total_speech_short / dur
-    speech_ratio_short = float("{:.2f}".format(speech_ratio_short))
-    number_of_pauses_short = len(silence_durations_short)
-    silence_durations_short = np.array(silence_durations_short)
-    std_short = np.std(silence_durations_short)
-    std_short = float("{:.2f}".format(std_short))
-    average_silence_dur_short = np.mean(silence_durations_short)
-    average_silence_dur_short = float("{:.2f}".format(average_silence_dur_short))
-    silence_seg_per_minute_short = float("{:.2f}".format(number_of_pauses_short/(dur/60.0)))
-    word_rate_in_speech_short = len(seg_limits_short)/total_speech_short
-    word_rate_in_speech_short = float("{:.2f}".format(word_rate_in_speech_short))
-
+    silence_features_short, number_of_pauses_short, total_speech_short = silence_features(seg_limits_short, dur)
     # long windows
-    total_speech_long = 0.0
-    counter = 0
-    silence_durations_long = []
-    for k in seg_limits_long:
-        if counter == 0:
-            silence_durations_long.append(k[0])
-        elif counter == (len(seg_limits_long) - 1):
-            p = counter - 1
-            silence_durations_long.append(k[0] - seg_limits_long[p][1])
-            silence_durations_long.append(dur - k[1])
-        else:
-            p = counter - 1
-            silence_durations_long.append(k[0] - seg_limits_long[p][1])
-        word_speech_long = k[1] - k[0]
-        total_speech_long = total_speech_long + word_speech_long
-        counter = counter + 1
-    speech_ratio_long = total_speech_long / dur
-    speech_ratio_long = float("{:.2f}".format(speech_ratio_long))
-    number_of_pauses_long = len(silence_durations_long)
-    silence_durations_long = np.array(silence_durations_long)
-    std_long = np.std(silence_durations_long)
-    std_long = float("{:.2f}".format(std_long))
-    average_silence_dur_long = np.mean(silence_durations_long)
-    average_silence_dur_long = float("{:.2f}".format(average_silence_dur_long))
-    silence_seg_per_minute_long = float("{:.2f}".format(number_of_pauses_long /
-                                                        (dur / 60.0)))
-    word_rate_in_speech_long = len(seg_limits_long) / total_speech_long
-    word_rate_in_speech_long = float("{:.2f}".format(word_rate_in_speech_long))
+    silence_features_long, number_of_pauses_long, total_speech_long = silence_features(seg_limits_long, dur)
+
 
     # B. segment model-based features
     # Load classifier:
+    dictionaries = []
+    for filename in os.listdir(models_directory):
+        if not (filename.endswith("MEANS")) and not (filename.endswith(".arff")):
+            model_path = os.path.join(models_directory, filename)
+            dictionary = predict_audio_labels(input_file, model_path)
```

Comment:
shouldn't this be 
`dictionary = predict_audio_labels(input_file, model_path)[0]`
?
Because predict_audio_labels also return "labels"